### PR TITLE
Exosuit Crash, Narrator and armor tweaks

### PR DESCRIPTION
**Pull request by NanakoAC**
Adds a major new ability to exosuits, 'Crash', allowing them to smash stuff at the cost of damage and power. They charge forward throwing their mass against things, and attempting to break through anything in their path. It hits everything solid in the tile until an object survives a hit, then stops. or it breaks everything and steps forward one tile. Using crash costs as much power as moving 20 normal steps
Mobs, and most pushable objects, are harmlessly shoved forward. They're only damaged if pinned against a wall. It deals 25-30 brute damage against mobs, but the 3 second cooldown makes it suboptimal as a melee attack.

Interestingly i did some calculations based on material costs yesterday, and even the lightest exosuit (Odysseus) weighs in at well over one metric ton. The crash ability is a step towards better reflecting the realities of driving a huge mass of metal around.

 In the process of adding this i tweaked ex_act interactions with a few major objects, notably walls, airlocks, windows, and girders, to make the behaviour more consistent. The most notable change among those is that explosions will generally deal cumulative damage if they fail to instantly destroy the object, so repeated explosions will take it down eventually. I also greatly buffed the health of high security doors (those on the vault, armory and AI core) to prevent making it too easy to break into secure areas. Previously these doors had the same health as all other airlocks, which seemed bizarre given their obviously sturdier build. And i made bolted doors take a bit less damage from crash and explosions, which gives an AI some means of slowing down an exosuit plowing through things. The logic being, running several thick steel bolts through an airlock is going to make it sturdier and more likely to survive impacts in general.

Crash is effective at forcing through most kinds of obstacles, including airlocks, firelocks, walls, grilles, windows, tables, etc.  It can break through reinforced walls, but is very inefficient at doing so, and it'll take a significant amount of time and hull integrity to break though. 

There are still a ton of less common objects it won't work on, like a lot of computers and machinery which are inexplicably indestructible to weaponry too. I plan to address those in a future projects focused on destruction. For now i've limited the scope of this PR.

Adds significant textual and audio feedback for low power and critical damage states, as well as some more voice lines that were sitting unused.

Plus a package of tweaks to armor values and EMP interactions with exosuits. Notably:
Ripley armor values changed to be more in line with the gygax, though slightly worse vs lasers and better vs brute damage. Logic here, the ripley is a heavily armored industrial suit, and the gygax is lightweight and designed for civilian law enforcement. Its like comparing a bulldozer and a police squad car.
Fixed some oversights with the firefighter ripley. It was only setting armor values on a few damagetypes, leaving the others blank. It now sets them all appropriately. I also reduced its speed to compensate for the generally higher durability
Buffed the durand a little too. It's the exosuit equivilant of a battle tank
Reduced the strength of EMP against exosuits generally, because it was a trivially overeffective weapon. But more interestingly, i made the EMP power drain based on flat values instead of a percentage of the power cell. Meaning that more advanced power cells offer a greater barrier against EMP, and thus encourage upgrading a bit more.

Also i did a spot of audio editing to remove an artefact from one of the voice files, which is explains the modification to nominalnano.ogg

changes: 
  - rscadd: "Added a Crash ability to exosuits. Uses the suit's mass to attempt to break through obstacles, sustaining some damage in the process."
  - tweak: "Buffed Ripley exosuit armor values significantly, and durand armour slightly. Firefighter ripley also buffed, but is slower than base ripley."
  - tweak: "Greatly increased the health values of some high-security airlocks."
  - bugfix: "Airlocks, tables, girders and windows now behave a bit more consistently when exploded. All airlocks are a bit more resistant to explosions when bolted."
  - rscadd: "Added some more narrator voices to exosuits."
  - rscadd: "Added some warning sounds for exosuits when low on power, or badly damaged."
  - tweak: "The power drain of EMPs used on exosuits no longer scales with the cell. A better power cell can now survive more EMP hits. Drain level is a little lower for the starting cell."
  - tweak: "EMP damage against exosuits reduced by 20%"

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -924,6 +938,154 @@
...
+	else
+		take_damage(damage*0.5)//An object that breaks hurts less than one that resists the impact
+
+	return 1
+
+
+/obj/mecha/proc/crash_damage(var/A)
+	if (istype(A, /mob/living))
+		var/mob/living/M = A
+		return min((M.mob_size / 3),2)//Crashing into a cow or cyborg hurts more than crashing into a dog
```
Comment (Bedshaped) [inline code/game/mecha/mecha.dm]: Surely being hit by a mech would do enormous damage and have a chance to do broken bones. Griefing wouldn't be a problem if you had admin logs.

Comment (NanakoAC) [inline code/game/mecha/mecha.dm]: The line you've commented here calculates the damage dealt **to the exosuit** when it hits a mob.
Mob damage is decided in that mob's ex_act function. Crashing into a mob just calls ex_act(3) on it

i didn't touch the damage dealt to mobs, but the 25-30 value that showed up in testing seemed fine. That will down someone in ~four hits and break plenty of bones

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -8,6 +8,13 @@
 #define RANGED 2
 
 
+#define NOMINAL		0
+#define FIRSTRUN	1
```
Comment (skull132) [inline code/game/mecha/mecha.dm]: #undef these at the bottom of the file.

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -924,6 +938,169 @@
...
+		var/mob/living/M = A
+		occupant.attack_log += "\[[time_stamp()]\]<font color='red'> Crashed into [key_name(M)]with exosuit [name] </font>"
+		M.attack_log += "\[[time_stamp()]\]<font color='orange'> Was rammed with the exosuit [name] driven by [key_name(occupant)]</font>"
+		msg_admin_attack("[key_name(occupant)] driving [name] crashed into [key_name(M)] at (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[M.x];Y=[M.y];Z=[M.z]'>JMP</a>)" )
+
+	A.ex_act(3)
+
+	sleep(1)
+	if (A && !(A.gcDestroyed) && A.type == oldtype)//We check if the object has been qdel'd or (for turfs) changed type
+		src.visible_message("<span class='danger'>[src.name] crashes into the [aname]</span>")
```
Comment (skull132) [inline code/game/mecha/mecha.dm]: Missing punctuation.

**Comment by NanakoAC** [comment]
Missing ! added

**Pull request merged**